src/lib/viewers/controls/settings/__tests__/SettingsDropdown-test.tsx

```diff
@@ -17,206 +15,115 @@ describe('SettingsDropdown', () => {
-            expect(mockEvent.stopPropagation).toHaveBeenCalled();
+            expect(screen.queryByTestId('bp-settings-flyout')).not.toHaveClass('bp-is-open');
         });
 
-        test.each(['ArrowUp', 'ArrowDown', 'Escape'])('should prevent propagation of keydown events for %s', key => {
-            const mockEvent = { key, stopPropagation: jest.fn() };
-            const wrapper = getWrapper();
-
-            // Open the flyout
-            act(() => {
-                wrapper.find('.bp-SettingsDropdown-button').simulate('click');
-            });
-            wrapper.update();
+        test.each(['ArrowUp', 'ArrowDown', 'Escape'])(
+            'should prevent propagation of keydown events for %s',
+            async key => {
+                const mockGlobalOnPress = jest.fn();
+                document.addEventListener('keydown', mockGlobalOnPress);
+                renderView();
 
-            act(() => {
-                wrapper.find(SettingsList).simulate('keydown', mockEvent);
-            });
-            wrapper.update();
+                await userEvent.click(screen.getByLabelText('Dropdown Label first'));
+                await userEvent.keyboard(`{${key}}`);
 
-            expect(mockEvent.stopPropagation).toHaveBeenCalled();
-        });
+                expect(mockGlobalOnPress).not.toHaveBeenCalled();
+                document.removeEventListener('keydown', mockGlobalOnPress);
+            },
+        );
     });
 
-    describe('ref', () => {
-        const TestComponent = (): JSX.Element => {
-            const ref = React.useRef<SettingsDropdownRef | null>(null);
-
-            React.useEffect(() => {
-                if (ref.current) {
-                    ref.current.focus();
-                }
-            }, []);
-            return <SettingsDropdown ref={ref} {...getDefaults()} />;
-        };
-
+    describe('focus', () => {
         test('should be able to focus on the dropdown button', () => {
-            const wrapper = mount(<TestComponent />, {
-                attachTo: getHostNode(),
-            });
+            renderView();
 
-            expect(wrapper.find('.bp-SettingsDropdown-button').getDOMNode()).toHaveFocus();
-        });
-    });
+            fireEvent.focus(screen.getByRole('button', { name: /first/i }));
 
-    describe('render', () => {
-        test('should return a valid wrapper', () => {
-            const wrapper = getWrapper();
-            const element = wrapper.getDOMNode();
-
-            expect(element).toHaveClass('bp-SettingsDropdown');
-            expect(wrapper.find('.bp-SettingsDropdown-label').text()).toBe('Dropdown Label');
-            expect(wrapper.find('.bp-SettingsDropdown-button').text()).toBe('first');
-            expect(wrapper.exists(SettingsFlyout));
-            expect(wrapper.exists(SettingsList));
+            waitFor(() => {
+                expect(document.activeElement).toBe(screen.getByRole('button', { name: /first/i }));
+            });
```

Comment:
Maybe to properly test it we should actual user event like click that focuses on element as well? 
If you await for such event then `waitFor` might not be needed?

Reply:
I agree. Currently you test (native) focus implementation.
Instead, I'd do sth like in this [example](https://testing-library.com/docs/user-event/convenience/): 

```
 test('should be able to focus on the dropdown button', async() => {
            const user = userEvent.setup()
            renderView();

            const button = screen.getByRole('button', { name: /first/i });

           await user.tab() 

           expect(button).toHaveFocus();
        });
```